app/controllers/ExemptionsForSmallProducersController.scala

```diff
@@ -65,12 +65,26 @@ private val form = formProvider()
           Future.successful(BadRequest(view(formWithErrors, mode))),
 
         value => {
-          val updatedUserAnswers = request.userAnswers.set(
-            ExemptionsForSmallProducersPage, value)
-
-          updateDatabaseAndRedirect(updatedUserAnswers, ExemptionsForSmallProducersPage, mode)
+          if (!value) {
+            updateSmallProducerList(request.userAnswers).map(updatedAnswersFinal =>
+              Redirect(navigator.nextPage(ExemptionsForSmallProducersPage, mode, updatedAnswersFinal)))
+          } else {
+            val updatedUserAnswers = request.userAnswers.set(ExemptionsForSmallProducersPage, value)
+            updateDatabaseAndRedirect(updatedUserAnswers, ExemptionsForSmallProducersPage, mode)
+          }
         }
       )
   }
 
+      private def updateSmallProducerList(userAnswers: UserAnswers): Future[UserAnswers] = {
+        for {
+          updatedAnswers <- Future.fromTry(userAnswers.set(ExemptionsForSmallProducersPage, false))
+          emtpySmallProducerList = updatedAnswers.smallProducerList.empty
```

Comment:
also rather than this function updating the db. couldnt it just return an updated user answers. and then on L70, you could call the updateDatabaseAndredirect method

Reply:
Extra line removed.

Reply:
Regarding getting rid of the extra function - this is used so that the page is updated and the small producer list is updated.